src/climate_assessment/checks.py

```diff
@@ -369,7 +369,7 @@ def check_reported_co2(df, filename, output_csv=False, outdir="output"):
         elif has_co2_total and has_co2_afolu:
             df_scen = _check_difference(df_scen, co2_total, co2_afolu, scen, model)
 
-        if co2_energy not in df_scen.variable:
+        if co2_energy not in df_scen._data.index.get_level_values("variable").unique():
```

Comment:
I think you'll need to make this replacement everywhere that `df_scen.variable` appears to get things to work properly.